### PR TITLE
Set up docs for deployment

### PR DESCRIPTION
**Pull request by MoritzWillmann**
### This PR
1. fixes all warnings regarding docs i.e.:
    - renames template folder
    - renames static folder from `images` to `static_`
    - removes unused examples from templates
    - removes autsectionlabel --> adds manual labels where needed
    - removes property members from method list --> they are in Attributes when needed
    - removes `ExpectationOperatorBase` and `ExpectationOperatorDerivatives` from `__init__.py`
    - adds `output="mpl"` to all `draw` calls in plot directives
    - fixes link to qiskit docs
    - fixes citation in docstring
    - adds blanklines before math directives
    - fixes variance operator in `VarianceLoss`
2. extends make clean to remove generated stuff
3. adds headers for Attributes sections in docstrings
4. adds a GitHub workflow that checks the docs on PR

### CODE REVIEW

File: src/squlearn/expectation_operator/__init__.py [hunk 1]
```diff
@@ -7,12 +7,7 @@
 )
 from .expectation_operator_implemented.ising_hamiltonian import IsingHamiltonian
 
-from .expectation_operator_derivatives import ExpectationOperatorDerivatives
-from .expectation_operator_base import ExpectationOperatorBase
-
 __all__ = [
-    "ExpectationOperatorBase",
-    "ExpectationOperatorDerivatives",
```
[the remark this answers is not in the document]
Comment (MoritzWillmann) [reply src/squlearn/expectation_operator/__init__.py:15]: Added to tomorrows agenda

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+matplotlib
```
[the remark this answers is not in the document]
Comment (MoritzWillmann) [reply docs/requirements.txt:1]: These are the requirements necessary to build the docs in the current state. If any docs need other extensions we need to add them here. This file is mainly for the action to be able to install it.

[the remark this answers is not in the document]
Comment (MoritzWillmann) [reply docs/requirements.txt:1]: > Where are the requirements installed in the github action, or is that not necessary?

This is done internally in the sphinx-action. It requires this file to be present.

[the remark this answers is not in the document]
Comment (David-Kreplin) [reply docs/requirements.txt:1]: My mistake, I have some other dependencies, but these are in branches that are not merged yet.

**Review by David-Kreplin — APPROVED** [review]
Looks great.

**Pull request merged**